wgpu-hal/src/vulkan/adapter.rs

```diff
@@ -1502,11 +1502,15 @@ impl crate::Adapter<super::Api> for super::Adapter {
                 .framebuffer_stencil_sample_counts
                 .min(limits.sampled_image_stencil_sample_counts)
         } else {
-            limits
-                .framebuffer_color_sample_counts
-                .min(limits.sampled_image_color_sample_counts)
-                .min(limits.sampled_image_integer_sample_counts)
-                .min(limits.storage_image_sample_counts)
+            match format.describe().sample_type {
+                wgt::TextureSampleType::Float { filterable: _ } => limits
+                    .framebuffer_color_sample_counts
+                    .min(limits.sampled_image_color_sample_counts),
+                wgt::TextureSampleType::Sint | wgt::TextureSampleType::Uint => {
+                    limits.sampled_image_integer_sample_counts
+                }
+                _ => unimplemented!(),
```

Comment:
```suggestion
                _ => unreachable!(),
```
I think this should be unreachable since we handle depth above.